R/rdeephaven/src/Makevars

```diff
@@ -1,64 +1,84 @@
 $(info The name of the shared library to be created is: $(SHLIB))
 rwildcard=$(foreach d,$(wildcard $(1:=/*)),$(call rwildcard,$d,$2) $(filter $(subst *,%,$2),$d))
 
-DEPENDENCY_DIRS = -L../lib/cpp-dependencies/local/abseil/lib \
-	-L../lib/cpp-dependencies/local/cares/lib \
-	-L../lib/cpp-dependencies/local/gflags/lib \
-	-L../lib/cpp-dependencies/local/grpc/lib \
-	-L../lib/cpp-dependencies/local/immer/lib \
-	-L../lib/cpp-dependencies/local/protobuf/lib \
-	-L../lib/cpp-dependencies/local/re2/lib \
-	-L../lib/cpp-dependencies/local/zlib/lib
+#
+# The code below assumes an environment variable DHCPP is defined and
+# points to a pre-existing directory where the deephaven C++ client
+# and its dependent libraries was built, as per the instructions in
+# https://github.com/deephaven/deephaven-core/blob/main/cpp-client/README.md
+#
 
+DEPENDENCY_DIRS = \
+	-L$(DHCPP)/local/re2/lib \
+	-L$(DHCPP)/local/zlib/lib \
+	-L$(DHCPP)/local/abseil/lib \
+	-L$(DHCPP)/local/cares/lib \
+	-L$(DHCPP)/local/gflags/lib \
+	-L$(DHCPP)/local/protobuf/lib \
+	-L$(DHCPP)/local/grpc/lib \
+	-L$(DHCPP)/local/arrow/lib
 
-DEPENDENCY_LIBS = -lclient -lprotobufd \
-	../lib/cpp-dependencies/local/arrow/lib/libarrow_flight.a \
-	../lib/cpp-dependencies/local/arrow/lib/libarrow.a \
-	../lib/cpp-dependencies/local/arrow/lib/libarrow_bundled_dependencies.a \
-	-lgrpc++ -lgrpc -laddress_sorting -lgpr -lupb -labsl_raw_hash_set -labsl_hashtablez_sampler -labsl_exponential_biased -labsl_hash -labsl_city \
-	-labsl_wyhash -labsl_statusor -labsl_bad_variant_access -lgpr -lupb \
-	-labsl_status -labsl_random_distributions -labsl_random_seed_sequences -labsl_random_internal_pool_urbg -labsl_random_internal_randen \
-	-labsl_random_internal_randen_hwaes -labsl_random_internal_randen_hwaes_impl -labsl_random_internal_randen_slow -labsl_random_internal_platform \
-	-labsl_random_internal_seed_material -labsl_random_seed_gen_exception -labsl_cord -labsl_bad_optional_access -labsl_str_format_internal \
-	-labsl_synchronization -labsl_stacktrace -labsl_symbolize -labsl_debugging_internal -labsl_demangle_internal -labsl_graphcycles_internal \
-	-labsl_malloc_internal -labsl_time -labsl_strings -labsl_throw_delegate -labsl_int128 -labsl_strings_internal -labsl_base -labsl_raw_logging_internal \
-	-labsl_log_severity -labsl_spinlock_wait -labsl_civil_time -labsl_time_zone -lssl -lre2 -lcares
+
+#
+# Note the name of the protobuf library changes to -lprotobuf
+# if the C++ client and its dependent libraries are built with
+#   CMAKE_BUILD_TYPE=Release
+# The value -lprotobufd matches the default build type `Debug`
+#
+DEPENDENCY_LIBS = \
+        -lclient \
+        -lprotobufd \
+        -larrow_flight -larrow -larrow_bundled_dependencies \
```

Comment:
I had to include the arrow libs as explicit `.a` files because in the previous build, they were also being built as `.so` files, so I needed to point to the static builds explicitly. Are we 100% sure that those libs are _not_ being built as shared libraries in the new build? If they are, the linker will prefer those and our thing will not work.